### PR TITLE
Create F2_22.c

### PR DESCRIPTION
**Pull request by italo-07**
The application you will be writing for is intended to run on a small, inexpensive embedded device that does not have a floating point unit. This means that the processor can only do integer arithmetic. However, the device does have a display and needs to display the result of floating point arithmetic to the user.  Because this is a small inexpensive processor there is no support for strings or any mathematical library functions. You must write all the code yourselves performing all the floating point math with integers only and no strings! Character arrays, or C strings, can't tell you their size but they do end with a '\0' character.

The requirements for the code are to write one of the functions presented below. All the functions return a boolean value indicating the success or failure of the operation. Improper data is the most likely cause for a function to fail. You may add your own additional helper functions (I had at least a half dozen) but you must include them for the review. Do not change the interface to these functions.

These two functions take as parameters two sets of mantissa and characteristic and a char array to hold the result of the arithmetic operations. The result of the add or subtract should be converted into char's and placed in the result array. The array must end with a '\0'. The 'len' parameter tells how many characters can be placed on the result array. For these functions to return true you must at a minimum store the characteristic of the result. If the result is a non-integer place as many of the digits as will fit in the result after a decimal point.

```c
bool add(int c1, int n1, int d1, int c2, int n2, int d2, char result[], int len);
bool subtract(int c1, int n1, int d1, int c2, int n2, int d2, char result[], int len); 
...
...
char answer[100];
int c1, n1, d1;
int c2, n2, d2;
 
c1 = 1;
n1 = 1;
d1 = 2;
 
c2 = 2;
n2 = 2;
d2 = 3; 
 
//if the C string could hold at least the characteristic if(add(c1, n1, d1, c2, n2, d2, answer, 100))
{
    //display string with answer 4.166666666...
}
else
{
    //display error message
}
'''

### CODE REVIEW

**Comment by kartikeyamani** [comment]
I am working on this

**Comment by kartikeyamani** [comment]
Hi,

After reviewing the code I've come up with some good points which improves code readabilty and efficiency

1. There are no comments for the code, adding comments improves readabilty of the code and it makes easy understanding for future developers.
2. Here there are no floating point numbers so we have to calculate integer part as well as fractional part and attach them and store as much as possible in result character array.
    I suppose if we can use helper functions like
         - gcd():  to reduce fractions(if needed)
         - itoa(): Converts an integer into its character representation and appends it to the result character array.
         - append_fraction: Converts the fractional part of the number into its decimal representation and appends it to the result string
3. Since there is no floating point number we can do the process like converting into common denominator and performing arithemetic and string the result in the result in the result charcater array.
4. All computations will be done with integers, and the fractional part can be calculated using a form of fixed-point arithmetic.
5. Using of helper functions makes problem breakdown into smaller tasks and it's easy to understand also.
6. We must handle the case of denominator is zero or else the program strucks.
7. We can Add logic to handle negative results, ensuring the characteristic is correctly marked with a minus sign if needed.
8. we must dynamically adjust the number of decimal places based on how much space is left in the buffer in the length array.
9. We must add  More complex test cases, especially edge cases, would help ensure that the logic works correctly in all situations (e.g., when the result is zero, negative, or involves large numbers). 
10. The example given doesn't clearly show how the fractions (n1/d1 and n2/d2) are being processed. Ensure that both fractions are converted to a common denominator before performing the addition or subtraction to avoid errors when combining two fractions.

File: Create F2_22.c [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+These two functions take as parameters two sets of mantissa and characteristic and a char array to hold the result of the arithmetic operations. The result of the add or subtract should be converted into char's and placed in the result array. The array must end with a '\0'. The 'len' parameter tells how many characters can be placed on the result array. For these functions to return true you must at a minimum store the characteristic of the result. If the result is a non-integer place as many of the digits as will fit in the result after a decimal point.
+```
+bool add(int c1, int n1, int d1, int c2, int n2, int d2, char result[], int len);
+bool subtract(int c1, int n1, int d1, int c2, int n2, int d2, char result[], int len); 
+...
+...
+'''
+```
+char answer[100];
```
Comment (Srikanth9793) [inline Create F2_22.c:16]: This defines a 100-character answer character array, which will be utilized to hold the arithmetic operation result. Enough space is allotted for the outcome, which includes the null terminator '\0' necessary for a C string.
Instead of creating an array statically, we can allocate it dynamically at runtime, allowing for flexible and efficient memory usage.

File: Create F2_22.c [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+int c1, n1, d1;
```
Comment (Srikanth9793) [inline Create F2_22.c:17]: The variable names should be more descriptive for clarity. Instead of writing c1, n1, and d1. We can give more appropriate names. This naming convention enhances readability by clearly indicating that these variables represent the integer and fractional components of two floating-point numbers.

This way user can read the code easily.
